File: api_post_ads/src/models/usuario.js
const { DataTypes } = require("sequelize");
const sequelize = require("../config/db");

const Usuario = sequelize.define(
  "Usuario",
  {
    idUsuario: {
      type: DataTypes.INTEGER,
      field: "id_usuario",
      primaryKey: true,
      autoIncrement: true,
      allowNull: false,
    },
    googleId: {
      type: DataTypes.STRING,
      field: "google_id",
      allowNull: true,
      unique: true,
    },
    nomUsuario: {
      type: DataTypes.STRING,
      field: "nom_usuario",
      allowNull: false,
    },
    apeUsuario: {
      type: DataTypes.STRING,
      field: "ape_usuario",
      allowNull: true,
    },
    userName: {
      type: DataTypes.STRING,
      field: "user_name",
      allowNull: true,
      unique: true,
    },
    emailUsuario: {
      type: DataTypes.STRING,
      field: "email_usuario",
      allowNull: false,
      unique: true,
    },
    passUsuario: {
      type: DataTypes.STRING,
      field: "pass_usuario",
      allowNull: true,
    },
    fotoUsuario: {
      type: DataTypes.STRING,
      field: "foto_usuario",
      allowNull: true,
    },
    emailVerified: {
      type: DataTypes.BOOLEAN,
      field: "email_verificado",
      defaultValue: false,
      allowNull: false,
    },
    codigoRecuperacion: {
      type: DataTypes.STRING,
      field: "codigo_recuperacion",
      allowNull: true,
    },
    codigoExpiraEn: {
      type: DataTypes.DATE,
      field: "codigo_expira_en",
      allowNull: true,
    },
    idTipoUsuario: {
      type: DataTypes.INTEGER,
      field: "id_tipousuario",
      allowNull: false,
      defaultValue: 0,
    },
    lvigente: {
      type: DataTypes.BOOLEAN,
      field: "lvigente",
      allowNull: false,
      defaultValue: true,
    },
  },
  {
    tableName: "usuario",
    timestamps: false,
  }
);

module.exports = Usuario;

File: post_ads/lib/models/usuario.dart
class Usuario {
  final int idUsuario;
  final String? googleId;
  final String nomUsuario;
  final String? apeUsuario;
  final String? userName;
  final String emailUsuario;
  final String? passUsuario;
  final String? fotoUsuario;
  final bool emailVerified;
  final int idTipoUsuario;
  final bool lvigente;

  Usuario({
    required this.idUsuario,
    this.googleId,
    required this.nomUsuario,
    this.apeUsuario,
    this.userName,
    required this.emailUsuario,
    this.passUsuario,
    this.fotoUsuario,
    required this.emailVerified,
    required this.idTipoUsuario,
    required this.lvigente,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json['idUsuario'],
      googleId: json['googleId'],
      nomUsuario: json['nomUsuario'] ?? '',
      apeUsuario: json['apeUsuario'] ?? '',
      userName: json['userName'] ?? '',
      emailUsuario: json['emailUsuario'] ?? '',
      passUsuario: json['passUsuario'] ?? '',
      fotoUsuario: json['fotoUsuario'],
      emailVerified: json['emailVerified'] ?? false,
      idTipoUsuario: json['idTipoUsuario'] ?? false,
      lvigente: json['lvigente'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idUsuario': idUsuario,
      'googleId': googleId,
      'nomUsuario': nomUsuario,
      'apeUsuario': apeUsuario,
      'userName': userName,
      'emailUsuario': emailUsuario,
      'passUsuario': passUsuario,
      'fotoUsuario': fotoUsuario,
      'emailVerified': emailVerified,
      'idTipoUsuario': idTipoUsuario,
    };
  }
}
